lib/Dialect/Handshake/HandshakeOps.cpp

```diff
@@ -1065,7 +1051,7 @@ void BufferOp::print(OpAsmPrinter &p) {
   int size =
       (*this)->getAttrOfType<IntegerAttr>("size").getValue().getZExtValue();
   p << " [" << size << "]";
-  p << " " << (*this)->getAttrOfType<StringAttr>("bufferType").getValue();
+  bufferTypeAttr().print(p);
```

Comment:
You're using `stringifyEnum` some places and printing the attribute elsewhere. It'd be nice to standardize on printing the attribute if possible. Also, this method of printing an attribute isn't the style we use in this code base (IIRC).

Reply:
Yeah, this was a somewhat conscious decision, although in hindsight, what's going on here is not very obvious. For some reason, the TableGenerated classes for the Attrs define `print()` methods that print out a space character before printing out the actual value's string. I had attempted to use these `print()` methods wherever I could, since they seemed more direct, but that extra space is a bit annoying and this specific instance was the only place where we were already printing out a space in the first place.

I agree that this is not very obvious, so I think I can change this to stringify the enum, like the other places.

Reply:
Changed this instance back to a `stringifyEnum()` call that gets shoveled into the `p` in https://github.com/llvm/circt/pull/2901/commits/f32651a34f96ad3b6ab53f5ef25e886206eb79be.

I left all the other instances where we convert these things to strings alone, since I think that's the best we can do in all those cases.

One thing that I remembered just now as I was making these changes is that the ModportDirectionAttr is a bit different than the BufferTypeAttr, since ModportDirectionAttr is only used within a StructFieldAttr, not directly by an Op. Ops can access either the Enum form or the Attr form (e.g. Handshake's BufferOp can use `bufferType()` to get the Enum, or `bufferTypeAttr()` to get the Attr), but you can't get the Enum form of an Attr within a StructFieldAttr as easily.